doc/source/development/maintaining.rst

```diff
@@ -445,9 +445,9 @@ which will be triggered when the tag is pushed.
    - Set as the latest release: Leave checked, unless releasing a patch release for an older version
      (e.g. releasing 1.4.5 after 1.5 has been released)
 
-5. Upload wheels to PyPI::
-
-    twine upload pandas/dist/pandas-<version>*.{whl,tar.gz} --skip-existing
+5. Wheels are uploaded automatically by GitHub Actions
+   via **Trusted Publishing** when the GitHub *Release*
+   is published. No manual ``twine upload`` step is required.
```

Comment:
```suggestion
   is published. Do not run ``twine upload`` manually.
```

Your language makes it sounds like it's optional to run `twine upload`, but we actually don't want this (but perhaps you can instead say that `twine upload` is only for exceptional circumstances).